pkg/sql/pgwire/server.go

```diff
@@ -1043,6 +1043,19 @@ func parseClientProvidedSessionParameters(
 				return sql.SessionArgs{}, err
 			}
 			for _, opt := range opts {
+				// crdb:jwt_auth_enabled must be passed as an option in order for us to support non-CRDB
+				// clients. jwt_auth_enabled is not a session variable. We extract it separately here.
+				if strings.ToLower(opt.key) == "crdb:jwt_auth_enabled" {
+					b, err := strconv.ParseBool(opt.value)
+					if err != nil {
+						return sql.SessionArgs{}, pgerror.Wrapf(
```

Comment:
nit: we could preserve the original error (which already has a message like "cannot parse X as bool") here with:
```
pgerror.Wrapf(err, pgcode.InvalidParameterValue, "crdb:jwt_auth_enabled")
```

or otherwise, avoid the redundant pgcodes with
```
errors.Wrapf(
  pgerror.Newf(
    pgcode.InvalidParameterValue,
    "parameter crdb:jwt_auth_enabled must have a boolean value",
  ),
  "options",
)
```

the advantage of the first is that it will also say what the passed-in value is, which would make debugging easier.

Reply:
Ah good call. I adopted your first example